### DISCUSSION TITLE
To generate ER diagram with weak association

### DISCUSSION THREAD

**Discussion by aaru59** [schemacrawler/SchemaCrawler#479] [Q&A]
I am configured SchemaCrawler through Docker container. I am using oracle database. In my database 100 tables are present. Data is inserted through data pipeline so PK and FK are not defined explicitly on tables. I have created attributes.yaml file , in that file I have mentioned catalog, schema, name and column references for both referenced-table and referencing-table. In command I have given weak-association=true --attributes-file /tmp/file.png  . Diagram is created but its not showing weak association between tables.
So my question is that is it necessary to define primary key on tables to display weak relationship?

**Comment by sualeh** [comment]
@aaru59 - no, you do not need to define primary key on tables to display weak associations. 

I am not sure why you have SchemaCrawler reading a PNG file instead of a YAML file. I think you should read up on [Docker mounts](https://docs.docker.com/storage/volumes/) to be able to read a local file from a Docker container. I cannot provide you with Docker support in this forum.

**Reply by aaru59** [reply]
I have checked log. It's reading attributes.yaml file correctly. I have pasted part of log beow just for info.
For catalog I have given database name in attributes.yaml file. Anything is missing from my side please let me know.
SchemaCrawler is not displaying weak association in ER diagram.

INFO: Loaded command config
Jun 10, 2021 5:09:42 AM schemacrawler.tools.commandline.command.ExecuteCommand run
CONFIG:
attributes-file: /tmp/attributes.yaml
weak-associations: true
Jun 10, 2021 5:09:42 AM schemacrawler.tools.commandline.command.ExecuteCommand run
INFO: Setting up SchemaCrawler command <details>
Jun 10, 2021 5:09:42 AM schemacrawler.tools.commandline.command.ExecuteCommand run
CONFIG: schemacrawler.tools.options.OutputOptions@1bfe3203[
  inputEncodingCharset: UTF-8
  outputEncodingCharset: UTF-8
  outputFormatValue: png
  outputResource: /tmp/WKDG.png
  title:
]
Jun 10, 2021 5:09:42 AM schemacrawler.tools.executable.CommandRegistry configureNewCommand
INFO: Matched provider <schemacrawler.tools.command.text.diagram.DiagramCommandProvider@5e9456ae>
J

**Reply by sualeh** [reply]
@aaru59 - please upload the complete log to a file sharing service such as Dropbox. Please see https://www.schemacrawler.com/consulting.html for the level of detail needed for support issues.
